src/components/ProductListing/ProductListing.jsx

```diff
@@ -2,10 +2,38 @@ import "./ProductListing.css";
 import { ProductCard } from "../ProductCard/ProductCard";
 import { useProduct } from "../../context/product-context";
 import { useEffect } from "react";
+import { actionTypes } from "../../constants/actionTypes";
 import axios from "axios";
+import {
+  getSortedProducts,
+  getCategorizedProducts,
+  getPriceRangedProducts,
+  getRatedProducts,
+} from "../../utilities/productFilters/index";
+
+const { LOAD_DATA } = actionTypes;
 
 const ProductListing = () => {
   const { productState, productDispatch } = useProduct();
+
+  const pricedProducts = getPriceRangedProducts(
+    productState.data,
+    productState.price
+  );
+  const ratedProducts = getRatedProducts(pricedProducts, productState.rating);
+  const categorizedProducts = getCategorizedProducts(
+    ratedProducts,
+    productState.categories
+    // productState.categories.category1,
+    // productState.categories.category2,
+    // productState.categories.category3,
```

Comment:
remove commented code if you not going to use it later on

Reply:
Yep I will remove it